src/app/dynamic/enterprise/quotas/component.ts

```diff
@@ -100,11 +103,41 @@ export class QuotasComponent implements OnInit {
     this.dataSource.filter = query;
   }
 
-  async add(): Promise<void> {
-    await lastValueFrom(
+  add(): void {
+    this._openDialog(null);
+  }
+
+  edit(quota: QuotaDetails): void {
+    this._openDialog(quota);
+  }
+
+  async delete({name}: QuotaDetails): Promise<void> {
+    const isConfirmed = await lastValueFrom(
```

Comment:
IMO, use this code snipper to avoid **3 reasons**

- Reason-1: Remove unnecessary async/await usages otherwise the return value always be promise due to async keyword
- Keep consistent with other components for `_matDaialog` subscriptions
- Yes async/await can be used INTERNALLY within subscription be part of Inner Callback (SCOPE)
- In case of future multiple calls can catered with `await` inside.

```suggestion
  delete({name}: QuotaDetails): void {
    this._matDialog
      .open<ConfirmationDialogComponent, ConfirmationDialogConfig, boolean>(ConfirmationDialogComponent, {
        data: {
          title: 'Delete Quota',
          message: `Delete quota for ${name}?`,
          confirmLabel: 'Delete',
        },
      })
      .afterClosed()
      .pipe(take(1))
      .subscribe(async (isConfirmed: boolean) => {
        if (!isConfirmed) return;
        await lastValueFrom(this._quotaService.deleteQuota(name));
        this._notificationService.success(`Deleting the ${name} datacenter`);
        this._quotaService.refreshQuotas();
      });
  }
```

Reply:
Using async await here is very inconsistent and ant-pattern. @Talha-Jamil-TJ can you please explain what was the reason for using it?

Reply:
The reason for using `async/await` is to make the code linear and easy to read, 
but I realise that I should propose this change first, 
so until then I'll revert back to using callbacks in the next PR.